commons-core/src/main/scala/com/avsystem/commons/serialization/json/JsonStringOutput.scala

```diff
@@ -17,8 +17,15 @@ object JsonStringOutput {
 trait BaseJsonOutput {
   protected final def indent(builder: JStringBuilder, indentSize: OptArg[Int], depth: Int): Unit =
     indentSize match {
-      case OptArg(size) => builder.append('\n').append(" " * (depth * size))
       case OptArg.Empty =>
+      case OptArg(size) =>
+        builder.append('\n')
+        val spaceNum = depth * size
+        var i = 0
```

Comment:
Looks like whitespace appending is implemented twice (here and in `writeField`). Can you extract a common method for that?